packages/loader/container-loader/src/container.ts

```diff
@@ -1311,10 +1309,8 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     }
 
     private async rehydrateDetachedFromSnapshot(detachedContainerSnapshot: ISummaryTree) {
-        const { snapshotTree, blobs } = getSnapshotTreeFromSerializedContainer(detachedContainerSnapshot);
-        blobs.forEach((value, key) => {
-            this.storageBlobs.set(key, value);
-        });
+        const snapshotTree = getSnapshotTreeFromSerializedContainer(detachedContainerSnapshot);
+        (this.storage as ContainerStorageAdapter).loadSnapshotForRehydratingContainer(snapshotTree);
```

Comment:
Any chance we can avoid casting here by having internal and external types? I.e. this._storage